compiler/lineinfos.nim

```diff
@@ -158,27 +163,15 @@ type
   TNoteKinds* = set[TNoteKind]
 
 const
-  NotesVerbosity*: array[0..3, TNoteKinds] = [
-    {low(TNoteKind)..high(TNoteKind)} - {warnShadowIdent, warnUninit,
-                                         warnProveField, warnProveIndex,
-                                         warnGcUnsafe,
-                                         hintSuccessX, hintPath, hintConf,
-                                         hintProcessing, hintPattern,
-                                         hintDependency,
-                                         hintExecuting, hintLinking,
-                                         hintCodeBegin, hintCodeEnd,
-                                         hintSource, hintStackTrace,
-                                         hintGlobalVar, hintGCStats},
-    {low(TNoteKind)..high(TNoteKind)} - {warnShadowIdent, warnUninit,
-                                         warnProveField, warnProveIndex,
-                                         warnGcUnsafe,
-                                         hintPath,
-                                         hintDependency,
-                                         hintCodeBegin, hintCodeEnd,
-                                         hintSource, hintStackTrace,
-                                         hintGlobalVar, hintGCStats},
-    {low(TNoteKind)..high(TNoteKind)} - {hintStackTrace, warnUninit},
-    {low(TNoteKind)..high(TNoteKind)}]
+  NotesVerbosity* = (proc(): array[0..3, TNoteKinds] =
```

Comment:
this change makes code more DRY, avoiding future bugs caused by someone putting a hint in 1 block of NotesVerbosity but not another block of NotesVerbosity

Reply:
That's obscure code, use named const sets to avoid the DRY violation.

Reply:
made the anonymous proc not anonymous (although i preferred the anonymous proc version, it's a common idiom n in C++, D, javascript + others, to avoid adding symbols in scope)

Reply:
That's not at all what I asked you to do. I asked you to use Nim's set feature. You know, that thing that C++, D, JS lack.